crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1791,6 +1791,9 @@ impl AuthorityPerEpochStore {
                 kind: ConsensusTransactionKind::CheckpointSignature(info),
                 ..
             }) => {
+                // We usually call notify_checkpoint_signature in SuiTxValidator, but that step can
+                // be skipped when a batch is already part of a certificate, so we must also
```

Comment:
nit: since this file is part of Sui, maybe specify these are Narwhal batch and Narwhal certificate.